src/app/WriteClient.cpp

```diff
@@ -269,7 +270,8 @@ CHIP_ERROR WriteClient::PutPreencodedAttribute(const ConcreteDataAttributePath &
         // list operation.
         // TODO (#38270): Generalize this behavior; send a non-empty ReplaceAll list for all clusters in a later Matter version and
         // enforce all clusters to support it in testing and in certification.
-        bool encodeEmptyListAsReplaceAll = !(path.mClusterId == Clusters::AccessControl::Id);
+        bool encodeEmptyListAsReplaceAll = !(path.mClusterId == Clusters::AccessControl::Id) ||
+            (testListEncodingOverride == TestListEncodingOverride::kForceLegacyEncoding);
```

Comment:
```suggestion
        bool encodeEmptyListAsReplaceAll = (path.mClusterId != Clusters::AccessControl::Id) ||
            (testListEncodingOverride == TestListEncodingOverride::kForceLegacyEncoding);
```

Using `!(a == b)` feels awkward ... why not `(a != b)` ?

Reply:
True! done!